app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -0,0 +1,149 @@
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+class ProfileChooserFragmentTest {
+
+  @Inject lateinit var profileTestHelper: ProfileTestHelper
+  @Inject lateinit var context: Context
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    setUpTestApplicationComponent()
+    profileTestHelper.initializeProfiles()
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerProfileChooserFragmentTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testProfileChooserFragment_initializeProfiles_checkProfilesAreShown() {
+    ActivityScenario.launch(ProfileActivity::class.java).use {
+      onView(atPositionOnView(R.id.profile_recycler_view, 0, R.id.profile_name_text)).check(matches(withText("Sean")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 0, R.id.profile_is_admin_text)).check(matches(withText(context.getString(R.string.profile_chooser_admin))))
+      onView(atPositionOnView(R.id.profile_recycler_view, 1, R.id.profile_name_text)).check(matches(withText("Ben")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 1, R.id.profile_is_admin_text)).check(matches(not(isDisplayed())))
+      onView(atPositionOnView(R.id.profile_recycler_view, 2, R.id.add_profile_text)).check(matches(withText(context.getString(R.string.profile_chooser_add))))
+    }
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testProfileChooserFragment_addManyProfiles_checkProfilesSortedAndNoAddProfile() {
+    profileTestHelper.addMoreProfiles(8)
+    ActivityScenario.launch(ProfileActivity::class.java).use {
+      onView(atPositionOnView(R.id.profile_recycler_view, 0, R.id.profile_name_text)).check(matches(withText("Sean")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 1, R.id.profile_name_text)).check(matches(withText("A")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 2, R.id.profile_name_text)).check(matches(withText("B")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 3, R.id.profile_name_text)).check(matches(withText("Ben")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 4, R.id.profile_name_text)).check(matches(withText("C")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 5, R.id.profile_name_text)).check(matches(withText("D")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 6, R.id.profile_name_text)).check(matches(withText("E")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 7, R.id.profile_name_text)).check(matches(withText("F")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 8, R.id.profile_name_text)).check(matches(withText("G")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 9, R.id.profile_name_text)).check(matches(withText("H")))
```

Comment:
This test case is failing, just because of one small point.
Add this `onView(withId(R.id.profile_recycler_view)).perform(scrollToPosition<RecyclerView.ViewHolder>(8))`

Meaning, you need to scroll to a particular item before writing test for it.

<img width="956" alt="Screenshot 2019-11-29 at 12 15 18 AM" src="https://user-images.githubusercontent.com/9396084/69827414-5cf3ad80-123d-11ea-8feb-9a78c6cdde70.png">